src/libutil/unix/file-descriptor.cc

```diff
@@ -120,14 +120,33 @@ void Pipe::create()
 
 //////////////////////////////////////////////////////////////////////
 
-void unix::closeMostFDs(const std::set<int> & exceptions)
+#if __linux__ or __FreeBSD__
```

Comment:
@rhelmot I only spotted the manpage on freebsd but does this also works in practice?